core-rust/state-manager/src/receipt.rs

```diff
@@ -64,15 +58,30 @@ pub struct DeletedSubstateVersion {
 
 #[derive(Debug, Clone, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
 pub enum LedgerTransactionOutcome {
+    Success,
+    Failure,
+}
+
+impl From<&TransactionOutcome> for LedgerTransactionOutcome {
```

Comment:
It's rare to see `From` implementations from references because the "From" trait suggests a conversion / ownership move, and actually using them is a bit messy - but this one looks like it might be okay 👍.

Alternatively, you can do this via a constructor on `LedgerTransactionOutcome` or a method on TransactionOutcome (although I guess we don't control that).

Reply:
I see; I guess this is mostly about the `From<>`'s convention.
I hope there is nothing wrong about resolving the `LedgerTransactionOutcome` from a `&TransactionOutcome`, so I simply renamed it to `LedgerTransactionOutcome::resolve(&TransactionOutcome)` 👍

Reply:
Sounds grand, thanks :)